server/server.go

```diff
@@ -212,6 +213,11 @@ func NewServer(opts *Options) (*Server, error) {
 	kp, _ := nkeys.CreateServer()
 	pub, _ := kp.PublicKey()
 
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = pub
```

Comment:
Hard to say the intended use is to have a more friendly identifier for monitoring so it’s not a critical thing. 

I co side red setting it to advertise name but that could be a LB or dns rr

I don’t really want to add a option or co figuration setting since it should do something sane for most people 

It still won’t be good enough should people put multiple servers on one machine but for cases like containers it gives a identifier that’s more usable than the PK when doing debugging. 

Hard to say what’s going to be the best option so I at least went for something that’s always there even when lookups fail and a useful value when not. 

The prom thing Colin works on will record this and server id so people can pick to use what they like and makes most sense.

Reply:
Can we not just use $HOST as addr in config if you want?

Reply:
What if someone just want it to listen on 127? Or on “::” so they get v4 and v6 etc? That’s the problem we have now we conflate the listen address and identity and it’s not good enough

Or maybe I don’t understand your suggestion. I don’t want everyone to have to craft specific confidant for monitoring to have names not public keys or all be 0.0.0.0

Reply:
Why nervous? It should for example change when the container gets recreated etc. For the purpose or specifically should not always be consistent :)

Reply:
Maybe we do this if not configured but let people override it via config as you suggest?

Reply:
Suggestions of what you'd like the new config item to be called? Once its configurable here then `hostname` is wrong so need something new

Reply:
Maybe just name or server_name and if you want you can set to $HOST (or whatever). Should be omitempty for purposes of varz.

Reply:
So again not following - I am saying we should use hostname when not set and this new `server_name` to override it.  So it would always be set? Why omitempty - we specifically need to make it so its always got some value :)

Reply:
I would argue that as server_name it must be set in config to show up, nothing automatic. And most configs may in fact set it to $HOST.